nbs/explains/background_tasks.ipynb

```diff
@@ -0,0 +1,296 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Simulated Slow API Service"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, create a very simple slow timestamp API. All it does is stall requests for a few seconds before returning JSON containing timestamps."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# slow_api.py\n",
+    "from fasthtml.common import *\n",
+    "from time import sleep, time\n",
+    "\n",
+    "app, rt = fast_app()\n",
+    "\n",
+    "@rt('/slow')\n",
+    "def slow(ts: int):\n",
+    "    sleep(3) # <1>\n",
+    "    return dict(request_time=ts, response_time=int(time())) # <2>\n",
+    "\n",
+    "serve(port=8123)\n",
+    "```\n",
+    "\n",
+    "1. This represents slow processing.\n",
+    "2. Returns both the task's original timestamp and the time after completion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Main FastHTML app"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's create a user-facing app that uses this API to fetch the timestamp from the glacially slow service."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# main.py\n",
+    "from fasthtml.common import *\n",
+    "from fastlite import *\n",
+    "from starlette.background import BackgroundTask\n",
+    "import time\n",
+    "import httpx\n",
+    "\n",
+    "app, rt = fast_app()\n",
+    "\n",
+    "db = database(':memory:')\n",
+    "\n",
+    "class TStamp: # <1>\n",
+    "    request_time: int # <2> \n",
+    "    response_time: int # <3>\n",
+    "\n",
+    "tstamps = db.create(TStamp, pk='request_time')\n",
+    "\n",
+    "def task_submit(request_time: int): # <4>\n",
+    "    client = httpx.Client()\n",
+    "    response = client.post(f'http://127.0.0.1:8123/slow?ts={request_time}') # <5>\n",
+    "    tstamps.insert(**response.json()) # <6>\n",
+    "\n",
+    "@rt\n",
+    "def submit():\n",
+    "    \"\"\"Route that initiates a background task and returns immediately.\"\"\"\n",
+    "    request_time = int(time.time())\n",
+    "    resp_content = P(f'Request submitted at: {request_time}')    \n",
+    "    task = BackgroundTask(task_submit, request_time=request_time) # <7>\n",
+    "    return FtResponse(resp_content, background=task) # <8>\n",
+    "\n",
+    "@rt\n",
+    "def show_tstamps(): return Ul(*[Li(t) for t in tstamps()]) # <9> \n",
+    "\n",
+    "@rt\n",
+    "def index():\n",
+    "    return Titled('Background Task Dashboard',\n",
+    "        P(Button('Press to call slow service', # <10> \n",
+    "            hx_post='/submit', hx_target='#res')),\n",
+    "        H2('Responses from Tasks'),\n",
+    "        P('', id='res'),\n",
```

Comment:
I think `map(Li, tstamps())` is nicer IMO - also I don't think you need `*` here IIRC.